madmom/ml/nn/layers.py

```diff
@@ -445,14 +611,27 @@ class GRULayer(Layer):
 
     """
 
-    def __init__(self, reset_gate, update_gate, cell, hid_init=None):
+    def __init__(self, reset_gate, update_gate, cell, init=None):
         # init the gates
         self.reset_gate = reset_gate
         self.update_gate = update_gate
         self.cell = cell
-        if hid_init is None:
-            hid_init = np.zeros(cell.bias.size, dtype=NN_DTYPE)
-        self.hid_init = hid_init
+        # keep the state of the layer
+        if init is None:
+            init = np.zeros(self.cell.bias.size, dtype=NN_DTYPE)
+        self.init = init
+        # keep the state of the layer
+        self._prev = self.init
+
```

Comment:
Missing a `__getstate__` here, because it's no longer in `Layer`. (I still think it's good to have it here instead of there)